meson.build

```diff
@@ -0,0 +1,85 @@
+project('jansson', 'c', version : '2.11', license : 'MIT')
+
+cc = meson.get_compiler('c')
+cdata = configuration_data()
+check_headers = [
+  ['HAVE_DLFCN_H', 'dlfcn.h'],
+  ['HAVE_ENDIAN_H', 'endian.h'],
+  ['HAVE_FCNTL_H', 'fcntl.h'],
+  ['HAVE_INTTYPES_H', 'inttypes.h'],
+  ['HAVE_LOCALE_H', 'locale.h'],
+  ['HAVE_MEMORY_H', 'memory.h'],
+  ['HAVE_SCHED_H', 'sched.h'],
+  ['HAVE_STDINT_H', 'stdint.h'],
+  ['HAVE_STDLIB_H', 'stdlib.h'],
+  ['HAVE_STRINGS_H', 'strings.h'],
+  ['HAVE_STRING_H', 'string.h'],
+  ['HAVE_SYS_PARAM_H', 'sys/param.h'],
+  ['HAVE_SYS_STAT_H', 'sys/stat.h'],
+  ['HAVE_SYS_TIME_H', 'sys/time.h'],
+  ['HAVE_SYS_TYPES_H', 'sys/types.h'],
+  ['HAVE_UNISTD_H', 'unistd.h'],
+]
+
+foreach h : check_headers
+  if cc.has_header(h.get(1))
+    cdata.set(h.get(0), 1)
+  endif
+endforeach
+
+check_functions = [
+# check token ('HAVE_ATOMIC_BUILTINS',)
+# check token ('HAVE_CLOSE',)
+  ['HAVE_GETPID', 'getpid', '#include<unistd.h>'],
+  ['HAVE_GETTIMEOFDAY', 'gettimeofday', '#include<sys/time.h>'],
+# check token ('HAVE_LOCALECONV',)
+# check token ('HAVE_LONG_LONG_INT',)
+# check token ('HAVE_OPEN',)
+# check token ('HAVE_READ',)
+# check token ('HAVE_SCHED_YIELD',)
+  ['HAVE_STRTOLL', 'strtoll', '#include<stdlib.h>'],
+# check token ('HAVE_SYNC_BUILTINS',)
+# check token ('HAVE_UNSIGNED_LONG_LONG_INT',)
+]
+
+foreach f : check_functions
+  if cc.has_function(f.get(1), prefix : f.get(2))
+    cdata.set(f.get(0), 1)
+  endif
+endforeach
+
+# FIXME
```

Comment:
Please expand FIXME explaining what is suboptimal here.

Reply:
Sounds fair. Will do.